gems/smithy/lib/smithy/anvil/client/templates/shapes.erb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+# This is generated code!
+
+module <%= namespace %>
+  module Shapes
+    include Smithy::Client::Shapes
+
+<%# Shapes definition -%>
+  <%- @shapes.each do |shape| -%>
+    <%- params = ["id: \"#{shape.id}\""] -%>
+    <%- params << "type: Types::#{shape.name}" if shape.typed -%>
+    <%- params << "traits: #{shape.traits}" unless shape.traits.empty? -%>
+    <%= "#{shape.name} = #{shape.shape_type}.new(" %><%= params.join(', ') %>)
+  <%- end -%>
+<%# Shape members definition %>
+  <%- shapes_with_members.each do |shape| -%>
+    <%- shape.members.each do |member| -%>
+    <%- params = [] -%>
+    <%- params << ["\"#{member.name}\""] unless %w[ListShape MapShape].include? shape.shape_type -%>
+    <%- params << [member.shape] -%>
+    <%- params << "traits: #{member.traits}" unless member.traits.empty? -%>
+    <%- case shape.shape_type -%>
+    <%- when 'ListShape' -%>
+    <%= "#{shape.name}.set_member(" %><%= params.join(', ') %>)
+    <%- when 'MapShape' -%>
+    <%= "#{shape.name}.set_member_#{member.name}(" %><%= params.join(', ') %>)
+    <%- else -%>
+    <%= "#{shape.name}.add_member(" %><%= params.join(', ') %>)
+    <%- end -%>
+    <%- end -%>
+  <%- end -%>
+<%# Schema definition %>
+    SCHEMA = Smithy::Client::Schema.new do |schema|
+      schema.service = ServiceShape.new(
+        id: "<%= @service_shape.id %>",
```

Comment:
Did you mean to reference the method? Let's not reference instance variables

Reply:
Yeah, I was going to fine-tune this